xarray/core/variable.py

```diff
@@ -928,21 +928,26 @@ def chunk(self, chunks=None, name=None, lock=False):
         else:
             if utils.is_dict_like(chunks):
                 chunks = tuple(chunks.get(n, s) for n, s in enumerate(self.shape))
-            # da.from_array works by using lazily indexing with a tuple of
-            # slices. Using OuterIndexer is a pragmatic choice: dask does not
-            # yet handle different indexing types in an explicit way:
-            # https://github.com/dask/dask/issues/2883
-            data = indexing.ImplicitToExplicitIndexingAdapter(
-                data, indexing.OuterIndexer
-            )
 
-            # For now, assume that all arrays that we wrap with dask (including
-            # our lazily loaded backend array classes) should use NumPy array
-            # operations.
-            if LooseVersion(dask.__version__) > "1.2.2":
-                kwargs = dict(meta=np.ndarray)
+            if not isinstance(data, np.ndarray) and hasattr(data, "__array_function__"):
+                # This should always be true after Variable.as_compatible_data
+                assert IS_NEP18_ACTIVE
+                kwargs = {}
             else:
-                kwargs = dict()
+                # da.from_array works by using lazily indexing with a tuple of
+                # slices. Using OuterIndexer is a pragmatic choice: dask does not
+                # yet handle different indexing types in an explicit way:
+                # https://github.com/dask/dask/issues/2883
+                data = indexing.ImplicitToExplicitIndexingAdapter(
```

Comment:
We should probably only do this wrapping if `data` is a subclass of `xarray.core.indexing.ExplicitlyIndexed`, which is a base class used for various backend arrays. For example, we should just let NumPy arrays through without wrapping. So this bit could just be:
```python
if isinstance(data, ExplicitlyIndexed):
    data = indexing.ImplicitToExplicitIndexingAdapter(
        data, indexing.OuterIndexer
    )
```

The reason why we have explicitly indexed arrays in the first place is so we can unambiguously handle array storage backends (like netCDF4 and h5py) that can't handle general array indexing. For example, in netCDF4 you can do "outer" indexing along two dimensions independent, which works differently from how NumPy handles it.

Xarray uses explicit indexing classes (`BasicIndexer`, `OuterIndexer` and `VectorizedIndexer`) which keep track of and canonicalize the desired indexing operation, without forcing everything into the [form desired by NumPy](https://docs.scipy.org/doc/numpy/reference/arrays.indexing.html) -- which has lots of edge cases (especially mixed `slice`/`ndarray` indexing) and isn't always easy to convert into other forms.